### ISSUE TITLE
Evaluate performance with different noise scenarios

### ISSUE THREAD

**Issue by tobias-liaudat** [CosmoStat/wf-psf#9]
It would be interesting to evaluate the model estimation performance with different noise scenarios.

Provide some quantitative metrics about the performance.

Try: 
- Different fixed SNRs. (ex. one fixed SNR for all the stars in the field)
- A field with stars with SNR in a certain range.

**Comment by tobias-liaudat** [comment]
This [commit](https://github.com/tobias-liaudat/wf-psf/commit/954c0a2d023bb9d9a2a24774d547beb9ee4079a8) includes a Colab notebook that includes the basic estimation of a noiseless dataset.

It provides a nice starting point for the PSF model.

**Comment by tobias-liaudat** [comment]
Here I'm leaving some results on datasets with different SNR.

### Description

The training of the model takes less than `4min` in GPU. Right, now the OPD has dimension `256x256`, the output PSF `64x64`, the dataset has `140` training PSFs and `60` testing PSFs (these last PSFs are not using to estimate the model). We are using only `15` Zernike coefficients and the number of wavelength bins is set to `15`.

I used the Adam optimiser (learning rate =` 1e-2`), the loss function is only a `l2 norm` over the reconstructions. The batch size is of `16` and the number of epochs `20`.

The notebook to reproduce the results is the one pushed in this [commit](https://github.com/tobias-liaudat/wf-psf/commit/ea282e2ef7140317c5ccf6a90129dc63cbe2377a).

Examples of the different PSFs looks like:

<img width="900" alt="Screenshot 2021-01-26 at 18 40 53" src="https://user-images.githubusercontent.com/16106854/105885018-df762980-6008-11eb-8d18-6ee3a02b9d7c.png">

### Result table 

The result in terms of RMSE is presented in the following table. The second column describes the pixel RMSE computed on the training dataset while the third column on the testing dataset. 

The first row shows the training without added noise. The varying SNR experience shows a dataset where the SNR of each observed star is drawn from a uniformly distributed random variable in the range [10, 70].

| SNR      | train RMSE | test RMSE  |
| :------: | :--------: | :--------: |
| -        | 1.7000e-04 | 1.6895e-04 |
| 10       | 1.4269e-04 | 1.4094e-04 |
| 20       | 1.9035e-04 | 1.8169e-04 |
| 30       | 2.0591e-04 | 2.0303e-04 |
| 50       | 2.2996e-04 | 2.3486e-04 |
| 70       | 1.9754e-04 | 1.9153e-04 |
| varying  | 1.6357e-04 | 1.5849e-04 |

## Example for SNR 10

Here I leave some example images from the testing dataset of SNR 10.

### Loss function

<img width="846" alt="Screenshot 2021-01-26 at 18 34 29" src="https://user-images.githubusercontent.com/16106854/105885346-514e7300-6009-11eb-907f-0b90674d0321.png">

### PSF recovery in sqrt() domain

<img width="780" alt="Screenshot 2021-01-26 at 18 33 58" src="https://user-images.githubusercontent.com/16106854/105885387-5e6b6200-6009-11eb-8240-c528e33e66a3.png">

### Matrix coefficient estimation

Each line represents a Zernike coefficient order and each column represents one of the monomial coefficient for the polynomial variations.

<img width="1105" alt="Screenshot 2021-01-26 at 18 34 11" src="https://user-images.githubusercontent.com/16106854/105885429-675c3380-6009-11eb-8a2b-ac7cd314021d.png">

**Comment by tobias-liaudat** [comment]
## Varying SNR example

I'm adding here another example with some supplementary plots.

Seems that the Zernike coefficients are estimated pretty well in the noisy scenario.

### Loss function

![Screenshot 2021-01-27 at 11 25 12](https://user-images.githubusercontent.com/16106854/105981598-8bb22180-6096-11eb-9bb8-85e586670940.png)

### PSF recovery in sqrt() domain

![Screenshot 2021-01-27 at 11 26 05](https://user-images.githubusercontent.com/16106854/105981635-97054d00-6096-11eb-8ab4-61fa3913dc32.png)

### Matrix Coefficient estimation

Each line represents a Zernike coefficient order and each column represents one of the monomial coefficient for the polynomial variations.

![Screenshot 2021-01-27 at 11 25 49](https://user-images.githubusercontent.com/16106854/105981665-a1274b80-6096-11eb-8fe8-ad6935461b58.png)

### Zernike coefficient estimation 

Each line represents a recovered PSF and each column an order of the Zernike Polynomial.

![Screenshot 2021-01-27 at 11 52 10](https://user-images.githubusercontent.com/16106854/105981694-ac7a7700-6096-11eb-9e4d-5f3ff0525d3a.png)


![Screenshot 2021-01-27 at 11 51 55](https://user-images.githubusercontent.com/16106854/105981713-b0a69480-6096-11eb-9a46-c36dbc367a75.png)

**Comment by tobias-liaudat** [comment]
I will close this issue since studies with different noise scenarios are already being made.
